cmd/syncthing/gui.go

```diff
@@ -1023,6 +1026,81 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileByte struct {
+	name string
+	buf  bytes.Buffer
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var arrayFile []fileByte
+
+	jsonConfig, _ := json.MarshalIndent(getRedactedConfig(s), "", "  ")
+	arrayFile = addBufferFile(arrayFile, "config.json", jsonConfig)
+
+	jsonLog, _ := json.MarshalIndent(s.systemLog.Since(time.Time{}), "", "  ")
+	arrayFile = addBufferFile(arrayFile, "log.json", jsonLog)
+
+	jsonError, _ := json.MarshalIndent(s.guiErrors.Since(time.Time{}), "", "  ")
+	if string(jsonError) != "null" {
+		arrayFile = addBufferFile(arrayFile, "error.json", jsonError)
+	}
+
+	panicFiles, ok := checkPanicFile(baseDirs["config"])
+	if ok {
+		for _, f := range panicFiles {
+			panicF, _ := ioutil.ReadFile(f)
+			arrayFile = addBufferFile(arrayFile, filepath.Base(f), panicF)
+		}
+	}
+
+	versionPlatform, _ := json.MarshalIndent(map[string]string{
+		"version":  Version,
+		"codename": Codename,
+		// "longVersion": LongVersion,
+		"os":   runtime.GOOS,
+		"arch": runtime.GOARCH,
+	}, "", "  ")
+	arrayFile = addBufferFile(arrayFile, "versionPlatform.json", versionPlatform)
+
+	usageReportingData, _ := json.MarshalIndent(getusageReportingData(s, r), "", "  ")
+	arrayFile = addBufferFile(arrayFile, "usageReportingData.json", usageReportingData)
+
+	rawConf := s.cfg.RawCopy()
+	debugCurrent := rawConf.GUI.Debugging
+	if !debugCurrent {
+		debugUndebug(s, rawConf, true)
+	}
+
+	heapPprof := getPprof(s, "/rest/debug/heapprof")
```

Comment:
I wouldn't grab these from the API. Instead, look at what the corresponding API methods do and just copy the code. It's just one to three lines each to create the profiles directly into a buffer, avoiding the API calls and undebug etc. stuff.